csrank/discretechoice/generalized_nested_logit.py

```diff
@@ -92,12 +92,7 @@ def __init__(
         """
         self.logger = logging.getLogger(GeneralizedNestedLogitModel.__name__)
 
-        self.n_object_features = n_object_features
-        self.n_objects = n_objects
-        if n_nests is None:
-            self.n_nests = n_objects + int(n_objects / 2)
```

Comment:
This looks like a bug to me. According to the documentation and the implementation in `nested_logit_model.py` we should drop the `n_objects` summand.

@prithagupta do you agree? (you added this code originally)

Reply:
Indeed, that seems to be a bug. We should probably open an issue about this and fix it on a separate pull request.

Reply:
In nested logit, we are taking default number of nests to be
self.n_objects = n_objects
        if n_nests is None:
            self.n_nests = int(n_objects / 2)
        else:
            self.n_nests = n_nests
 , while in generalized_nested_logit,
its
  self.n_nests = n_objects + int(n_objects / 2)

The nested logit model divides the objects into a mutually exclusive set so the nests should be less than number of objects if n_nests=n_objects, then it becomes a logit model.
While GenNestedModel, says every object to some extend is a part of a nest, there is some literature which suggests still the number of nests should be less than number of objects in the choice set. So, in principle it is a good idea to get the number of objects in a choice set.